src/piece_picker.cpp

```diff
@@ -2248,6 +2249,29 @@ namespace {
 				if (num_blocks <= 0) return ret;
 			}
 		}
+		else if (options & priority_order)
+		{
+			if (m_dirty) update_pieces();
+			TORRENT_ASSERT(!m_dirty);
+
+			for (uint8_t prio = 7; prio > 0; --prio)
+			{
+				for (auto i : m_pieces)
```

Comment:
this is extremely inefficient. To the point where your download speed may be CPU bound. This is not the way to do this.

If you think priorities should strictly take precedence over rarity, change the order in the priority boundary to put all prio-7 pieces first, then prio-6 and so on.
The main challenge with that is that the availability of a piece is virtually unbounded, so you don't know what to multiply the priority by. You may have to set a cap to 100 or something.

Reply:
> If you think priorities should strictly take precedence over rarity, change the order in the priority boundary to put all prio-7 pieces first, then prio-6 and so on.

You mean to change  `m_priority_boundaries` fill-up logic? 
Trying to understand how it is done currently with rarest pick mode.